expan/core/binning.py

```diff
@@ -95,24 +96,28 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
-    def apply_to_data(self, data):
+    def apply_to_data(self, data, dimension):
         """
         Apply the bin to data.
-        :param data: a pandas Series
+        :param data: pandas data frame
+        :param dimension: dimension of the data for this bin
         :return: subset of input data which belongs to this bin
         """
+        data_subgroup_dimension = data[dimension]
+
         # if either bound is nan, only nans exist in the bin.
         if np.isnan(self.lower) or np.isnan(self.upper):
-            return data[np.isnan(data)]
+            return data[np.isnan(data_subgroup_dimension)]
 
         if self.lower_closed:
-            filter_lower = (data >= self.lower)
+            filter_lower = (data_subgroup_dimension >= self.lower)
         else:
-            filter_lower = (data > self.lower)
+            filter_lower = (data_subgroup_dimension > self.lower)
         if self.upper_closed:
-            filter_upper = (data <= self.upper)
+            filter_upper = (data_subgroup_dimension <= self.upper)
```

Comment:
just to clarify: is data_subgroup_dimension if the dimension of one subgroup? does it mean if there are 3 kpis then the data_subgroup_dimension will be 3?

Reply:
It will only be one column (the given feature name) of the data. 
-> then we compare this column with the defined bin value to find out what belongs to the subgroup 
-> results in a column of booleans
-> then we use these booleans to filter the complete dataframe again (here you will have a dataframe with kpi1, kpi2, and all other columns but only with rows belongs this subgroup)

Reply:
clear! thanks!